### ISSUE TITLE
Some messenger doesn't save their login on app restart

### ISSUE THREAD

**Issue by MartinX3** [sonnyp/Tangram#187]
**Tangram version**
1.4.0

**Linux distribution**
Arch Linux / EndeavourOS

**Flatpak version**
No flatpak.
Compiled via https://aur.archlinux.org/packages/tangram

**Desktop environment**
XFCE 4.16

I need to login again on some messengers after an app restart.

Messenger saving my login on app restart:
- Element / Matrix
- Slack
- Telegram WebZ

Messenger resetting my login on app restart:
- Zulip

**Comment by sonnyp** [comment]
Do you get an error in the terminal if you start Tangram with `flatpak run re.sonny.Tangram` ?

**Comment by MartinX3** [comment]
It can't find
- libhspell.so.0
- libnuspell.so.5
- libvoikko.so.1

And I get
`(process:70741): libsoup-WARNING **: 20:20:42.919: Can't open /home/$USER/.local/share/Tangram/53b606e2cbb841fd8bb539f2470e19b0/cookies.sqlite`

**Comment by sonnyp** [comment]
@MartinX3 can you open Tangra from the terminal, reproduce the issue and share the full logs?

**Comment by MartinX3** [comment]
```
$ re.sonny.Tangram 
Gjs-Message: 20:26:30.484: JS LOG: gjs 1.72.0
Gjs-Message: 20:26:30.484: JS LOG: WebKitGTK 2.36.0
Gjs-Message: 20:26:30.484: JS LOG: libsoup 2.74.2
Gjs-Message: 20:26:30.484: JS LOG: env: host
Gjs-Message: 20:26:30.484: JS LOG: data_dir: /home/martin/.local/share/Tangram
Gjs-Message: 20:26:30.484: JS LOG: cache_dir: /tmp/martin/.cache/Tangram
Gjs-Message: 20:26:30.484: JS LOG: config_dir: /home/martin/.config/Tangram
Gjs-Message: 20:26:30.484: JS LOG: applications_dir: /home/martin/.local/share/applications
Gjs-Message: 20:26:30.484: JS LOG: keyfile_settings_path: 
Gjs-Message: 20:26:30.492: JS LOG: programInvocationName: /usr/bin/re.sonny.Tangram
Gjs-Message: 20:26:30.492: JS LOG: _: /usr/bin/re.sonny.Tangram
Gjs-Message: 20:26:30.492: JS LOG: argv /usr/bin/re.sonny.Tangram

** (re.sonny.Tangram:76500): WARNING **: 20:26:30.570: Error loading plugin: libhspell.so.0: Kann die Shared-Object-Datei nicht öffnen: Datei oder Verzeichnis nicht gefunden


** (re.sonny.Tangram:76500): WARNING **: 20:26:30.571: Error loading plugin: libnuspell.so.5: Kann die Shared-Object-Datei nicht öffnen: Datei oder Verzeichnis nicht gefunden


** (re.sonny.Tangram:76500): WARNING **: 20:26:30.571: Error loading plugin: libvoikko.so.1: Kann die Shared-Object-Datei nicht öffnen: Datei oder Verzeichnis nicht gefunden


(process:76545): libsoup-WARNING **: 20:26:30.813: Can't open /home/martin/.local/share/Tangram/53b606e2cbb841fd8bb539f2470e19b0/cookies.sqlite
Gjs-Message: 20:26:31.032: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 20:26:31.032: JS LOG: navigation,OTHER,https://web.whatsapp.com/
Gjs-Message: 20:26:31.036: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 20:26:31.036: JS LOG: navigation,OTHER,https://discord.com/channels/@me
Gjs-Message: 20:26:31.095: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 20:26:31.095: JS LOG: navigation,OTHER,https://app.slack.com/client/hide/hide
Gjs-Message: 20:26:31.096: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 20:26:31.096: JS LOG: navigation,OTHER,https://app.element.io/index.html
Gjs-Message: 20:26:31.099: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 20:26:31.099: JS LOG: navigation,OTHER,https://twrp.zulipchat.com/
Gjs-Message: 20:26:31.099: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 20:26:31.099: JS LOG: navigation,OTHER,https://web.telegram.org/z/
Gjs-Message: 20:26:31.100: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 20:26:31.100: JS LOG: navigation,OTHER,https://hide.slack.com/ssb/redirect?entry_point=workspace_signin
Gjs-Message: 20:26:31.180: JS LOG: decide-policy,RESPONSE
Gjs-Message: 20:26:31.443: JS LOG: decide-policy,RESPONSE
Gjs-Message: 20:26:31.451: JS LOG: decide-policy,RESPONSE
Gjs-Message: 20:26:31.477: JS LOG: decide-policy,RESPONSE
Gjs-Message: 20:26:31.483: JS LOG: decide-policy,RESPONSE
Gjs-Message: 20:26:31.495: JS LOG: decide-policy,RESPONSE
Gjs-Message: 20:26:31.753: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 20:26:31.753: JS LOG: navigation,OTHER,https://twrp.zulipchat.com/login/

(process:76545): libsoup-WARNING **: 20:26:31.938: Can't open /home/martin/.local/share/Tangram/53b606e2cbb841fd8bb539f2470e19b0/cookies.sqlite
Gjs-Message: 20:26:31.964: JS LOG: decide-policy,RESPONSE
Gjs-Message: 20:26:32.083: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 20:26:32.084: JS LOG: navigation,OTHER,about:blank
Gjs-Message: 20:26:32.088: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 20:26:32.088: JS LOG: navigation,OTHER,slack://hide/magic-login/hide
Gjs-Message: 20:26:32.143: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 20:26:32.143: JS LOG: navigation,OTHER,about:blank
Gjs-Message: 20:26:32.145: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 20:26:32.145: JS LOG: navigation,OTHER,about:blank
Gjs-Message: 20:26:32.251: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 20:26:32.251: JS LOG: navigation,OTHER,about:srcdoc
Gjs-Message: 20:26:34.359: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 20:26:34.359: JS LOG: navigation,OTHER,https://app.element.io/index.html#/room/!hide:matrix.org
WebKit wasn't able to find a WebVTT encoder. Subtitles handling will be degraded unless gst-plugins-bad is installed.
GStreamer element fakevideosink not found. Please install it
WebKit wasn't able to find a WebVTT encoder. Subtitles handling will be degraded unless gst-plugins-bad is installed.
Gjs-Message: 20:26:37.106: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 20:26:37.106: JS LOG: navigation,OTHER,https://hide.slack.com/?entry_point=workspace_signin
Gjs-Message: 20:26:37.256: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 20:26:37.256: JS LOG: navigation,OTHER,https://hide.slack.com/messages
Gjs-Message: 20:26:37.384: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 20:26:37.384: JS LOG: navigation,OTHER,https://app.slack.com/client/hide
Gjs-Message: 20:26:37.549: JS LOG: decide-policy,RESPONSE
Gjs-Message: 20:26:37.998: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 20:26:37.999: JS LOG: navigation,OTHER,about:blank
Gjs-Message: 20:26:38.088: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 20:26:38.088: JS LOG: navigation,OTHER,about:srcdoc
```

**Comment by sonnyp** [comment]
I cannot reproduce with

1. Add a tab
2. Enter https://chat.zulip.org/
3. Login
4. Click "Done"
5. Restart Tangram

Zulip is stilled logged in

I see you edited the description and removed things from your original report. So it's not very clear anymore.

Please clarify the description, title, and steps to reproduce, specially if it only affects a single website for you.

**Comment by MartinX3** [comment]
I did not remove stuff from my original report. I just replaced the credential strings with custom values.
It's only Zulip here at https://twrp.zulipchat.com/

Do you get the same error message about the invalid path in your logs?

**Comment by sonnyp** [comment]
So what URLs exactly do you reproduce the problem with?

Can you reproduce with Flatpak Tangram?

> I get the same issue and now I have two error messages in my log.

It's just a warning and probably unrelated since this warning has existed for as long as Tangram did and nobody reported a similar issue so far https://github.com/sonnyp/Tangram/issues/57 . Good to know it's not Flatpak specific thought. 

> I did not remove stuff from my original report. I just replaced the credential strings with custom values.

You removed Discord and info about tmpfs. It's ok - I'm just trying to understand what the problem is exactly and how I can reproduce.

**Comment by MartinX3** [comment]
> So what URLs exactly do you reproduce the problem with?

I think chat.zulip.com as well, but mainly I use https://twrp.zulipchat.com/

> Can you reproduce with Flatpak Tangram?

If I remove the Tangram System package from my Arch Linux and install the Flatpak version, will it delete my saved chats?

> > I get the same issue and now I have two error messages in my log.
> 
> It's just a warning and probably unrelated since this warning has existed for as long as Tangram did and nobody reported a similar issue so far #57 . Good to know it's not Flatpak specific thought.

Oh, sad, I thought that was the reason, because it can't save its cookies and so forgetting my login.

> > I did not remove stuff from my original report. I just replaced the credential strings with custom values.
> 
> You removed Discord and info about tmpfs. It's ok - I'm just trying to understand what the problem is exactly and how I can reproduce.

I only removed stuff I can't repoduce anymore.
The discord warning disappeared and discord got its own folder with its cookie database after I think and now I don't loose my login.

**Comment by sonnyp** [comment]
> I think chat.zulip.com as well, but mainly I use https://twrp.zulipchat.com/

If you want this solved, you gotta help - please make sure.

> If I remove the Tangram System package from my Arch Linux and install the Flatpak version, will it delete my saved chats?

You don't need to remove anything and the flatpak will have its own data/cache folders so it won't interfere. 

```
flatpak install re.sonny.Tangram
flatpak run re.sonny.Tangram
```

backup `~/.local/share/Tangram` if you want

> The discord warning disappeared and discord got its own folder with its cookie database after I think and now I don't loose my login.


This is useful information that may help debug this.

If you find a way to reproduce the problem with Discord that would help.

Like deleting (don't forget to backup) `~/.local/share/Tangram`, and adding Discord again or something like that

**Comment by MartinX3** [comment]
Thank you for your help.
Discord works since the first and only occurrence of the reseted user credentials.

I tried to run `flatpak run re.sonny.Tangram` but sadly there is no window and I get the following error in the terminal:
```bash
$ flatpak run re.sonny.Tangram

Note that the directories 

'/var/lib/flatpak/exports/share'
'/home/martin/.local/share/flatpak/exports/share'

are not in the search path set by the XDG_DATA_DIRS environment variable, so
applications installed by Flatpak may not appear on your desktop until the
session is restarted.

Gjs-Message: 21:09:07.626: JS LOG: gjs 1.72.0
Gjs-Message: 21:09:07.626: JS LOG: WebKitGTK 2.36.0
Gjs-Message: 21:09:07.626: JS LOG: libsoup 2.74.2
Gjs-Message: 21:09:07.626: JS LOG: env: flatpak
Gjs-Message: 21:09:07.626: JS LOG: data_dir: /home/martin/.var/app/re.sonny.Tangram/data/Tangram
Gjs-Message: 21:09:07.626: JS LOG: cache_dir: /home/martin/.var/app/re.sonny.Tangram/cache/Tangram
Gjs-Message: 21:09:07.626: JS LOG: config_dir: /home/martin/.var/app/re.sonny.Tangram/config/Tangram
Gjs-Message: 21:09:07.626: JS LOG: applications_dir: /home/martin/.local/share/applications
Gjs-Message: 21:09:07.626: JS LOG: keyfile_settings_path: 
Gjs-Message: 21:09:07.635: JS LOG: programInvocationName: /app/bin/re.sonny.Tangram
Gjs-Message: 21:09:07.635: JS LOG: _: /usr/bin/flatpak
Gjs-Message: 21:09:07.636: JS LOG: argv /app/bin/re.sonny.Tangram
Gtk-Message: 21:09:07.636: Failed to load module "xapp-gtk3-module"
Gtk-Message: 21:09:07.636: Failed to load module "xapp-gtk3-module"
Gtk-Message: 21:09:07.636: Failed to load module "canberra-gtk-module"
Gtk-Message: 21:09:07.636: Failed to load module "canberra-gtk-module"

(re.sonny.Tangram:2): Gjs-CRITICAL **: 21:09:07.729: JS ERROR: Gio.IOErrorEnum: Fehler beim Erstellen des Ordners »/home/martin/.var/app/re.sonny.Tangram/data/Tangram/718138181c024542b8fdc3212e2f40ab«: Datei oder Verzeichnis nicht gefunden
ensureDirectory@file:///app/share/Tangram/util.js:107:33
create@file:///app/share/Tangram/instances.js:70:18
onNewTab@file:///app/share/Tangram/window.js:226:36
Window/<@file:///app/share/Tangram/window.js:254:7
observeSetting@file:///app/share/Tangram/util.js:71:9
Window@file:///app/share/Tangram/window.js:251:17
getWindow@file:///app/share/Tangram/application.js:24:14
@file:///app/share/Tangram/application.js:35:3
main@file:///app/share/Tangram/main.js:59:22
@file:///app/bin/re.sonny.Tangram:18:5
```

**Comment by MartinX3** [comment]
Ah, I see now that you released v1.4.1.
I updated and the `Can't open` errors disappeared.
Zulip works now.
Thank you.

**Comment by gregorydk** [comment]
Discord still doesn't save login, using flatpak Tangram 2.0.
